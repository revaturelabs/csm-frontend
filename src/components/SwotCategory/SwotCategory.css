.item {
    background-color: white !important;
    border-radius: 0.5rem !important;
    text-align: left;
    padding: .5rem !important;

}