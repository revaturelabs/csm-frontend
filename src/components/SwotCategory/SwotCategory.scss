@import "../../styles/main";

.item {
    background-color: white !important;
    border-radius: 0.3rem !important;
    text-align: left;
    padding: .5rem !important;
}

.ui.button#edit {
    color: white;
    background-color: $revature-blue;
}

.ui.button#edit:hover {
    background-color:rgb(100, 150, 180);
}

div.description > button {
    transition: 0s ease .5s;
    &#edit, &#delete { display: none; }
}

div.description:hover > button {
    transition: 0s ease .5s;
    &#edit, &#delete { display: inline-block; }
}