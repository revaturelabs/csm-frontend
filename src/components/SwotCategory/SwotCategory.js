import React from "react";
import { useDispatch } from "react-redux";
import { Button, Item } from "semantic-ui-react";
import "./SwotCategory.css";

/**
 *  Representation for a single category and it's associated notes in the SWOT Analysis Widget
 * @param {Object} props  The values passed to the component by its parent
 * @param {string} props.category The category being displayed by this SwotCategory Instance
 * @param {string} props.note The note associated with the category
 * @param {CallableFunction} props.editHandler The function to be called when the edit button is clicked
 * @param {CallableFunction} props.deleteHandler The function to be called when the delete button is clicked
 */
const SwotCategory = (props) => {
  const dispatch = useDispatch();

  /**
   * Handles setting the store to contain the current category and calling the appropriate fn
   * Based on the id of the event that has been passed in
   * Functions are passed through props by the SwotQuadrantTable, the parent to this component
   * @param {Event} event The button that was pressed to trigger the method call
   */
  const handlerWrapper = (event) => {
    event.preventDefault();
    event.stopPropagation();
    // Set the store to have the correct category context
    dispatch({ type: "updateCategory", category: props.category });

    // Call the appropriate handler as determined by what button triggered this fn call
    if (event.target.id === "edit") {
      props.editHandler(props.category);
    } else {
      props.deleteHandler(props.key);
    }
  };
  /**
   * A SwotCategory is a wrapper of a Semantic UI Item component.
   * It contains 4 things:
   * 1. The name of the category
   * 2. Notes for the category
   * 3. An Edit button
   * 4. A Delete Button
   */
  return (
    <Item>
      <Item.Content>
        <Item.Header>{props.category}</Item.Header>
        <Item.Meta>Notes</Item.Meta>
        <Item.Description>
          <p>{props.note}</p>
          <br></br>
          <Button color="blue" id={"edit"} onClick={handlerWrapper}>
            Edit
          </Button>
          <Button color="red" id={"delete"} onClick={handlerWrapper}>
            Delete
          </Button>
        </Item.Description>
      </Item.Content>
    </Item>
  );
};

export default SwotCategory;
