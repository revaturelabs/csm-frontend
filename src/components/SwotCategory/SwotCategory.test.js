// Component.test.js
import React from "react";
import ReactDOM from "react-dom";
import Enzyme, { shallow, mount, render, fireEvent } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { useSelector, useDispatch } from 'react-redux';
import configureStore from 'redux-mock-store';
import reducer from '../../reducers';
import SwotCategory from './SwotCategory';

jest.mock('react-redux');

describe("Swot Category unit tests", () => {

    it('Rendering component without props.', () => {
        const component = shallow(<SwotCategory/>);
        expect(component).toMatchSnapshot();
    });

    it('Rendering component with props for category and note.', () => {
        const category="TEST VALUE";
        const note='Test note for testing';
        const component = shallow(<SwotCategory category={category} note={note} />);
        expect(component).toMatchSnapshot();
    });

    it('Handles calling the proper handler following a delete button press', () => {
        const category="TEST VALUE";
        const note='Test note for testing';
        const mockEditHandler=jest.fn();
        const mockDeleteHandler = jest.fn();
        const component = mount(<SwotCategory category={category} note={note} editHandler={mockEditHandler} deleteHandler={mockDeleteHandler}></SwotCategory>)
        component.unmount();

    })

})
