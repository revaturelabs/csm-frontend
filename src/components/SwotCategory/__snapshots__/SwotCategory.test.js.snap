// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swot Category unit tests Rendering component with props for category and note. 1`] = `
<Item>
  <ItemContent>
    <ItemHeader>
      TEST VALUE
    </ItemHeader>
    <ItemMeta>
      Notes
    </ItemMeta>
    <ItemDescription>
      <p>
        Test note for testing
      </p>
      <br />
      <Button
        as="button"
        color="blue"
        id="edit"
        onClick={[Function]}
      >
        Edit
      </Button>
      <Button
        as="button"
        color="red"
        id="delete"
        onClick={[Function]}
      >
        Delete
      </Button>
    </ItemDescription>
  </ItemContent>
</Item>
`;

exports[`Swot Category unit tests Rendering component with props that match the original component. 1`] = `
<Item>
  <ItemContent>
    <ItemHeader>
      TEST VALUE
    </ItemHeader>
    <ItemMeta>
      Notes
    </ItemMeta>
    <ItemDescription>
      <p>
        Test note for testing
      </p>
      <br />
      <Button
        as="button"
        color="blue"
        id="edit"
        onClick={[Function]}
      >
        Edit
      </Button>
      <Button
        as="button"
        color="red"
        id="delete"
        onClick={[Function]}
      >
        Delete
      </Button>
    </ItemDescription>
  </ItemContent>
</Item>
`;

exports[`Swot Category unit tests Rendering component without props. 1`] = `
<Item>
  <ItemContent>
    <ItemHeader />
    <ItemMeta>
      Notes
    </ItemMeta>
    <ItemDescription>
      <p />
      <br />
      <Button
        as="button"
        color="blue"
        id="edit"
        onClick={[Function]}
      >
        Edit
      </Button>
      <Button
        as="button"
        color="red"
        id="delete"
        onClick={[Function]}
      >
        Delete
      </Button>
    </ItemDescription>
  </ItemContent>
</Item>
`;
